packages/web/src/generators/application/application.ts

```diff
@@ -298,7 +298,7 @@ function normalizeOptions(host: Tree, options: Schema): NormalizedSchema {
     ? options.tags.split(',').map((s) => s.trim())
     : [];
 
-  if (options.bundler === 'vite') {
+  if (options.bundler === 'vite' && !options.unitTestRunner) {
```

Comment:
Shouldn't `unitTestRunner` default to jest when the bundler is `webpack/none` ?
(As documented in schema.json below)

Reply:
that's technically a bad check as unitTestRunner will never be falsey, since it's always defaulted in normalizeOptions
https://github.com/nrwl/nx/blob/master/packages/web/src/generators/application/application.ts#L301-L307

also have test cases for the bundler options.
https://github.com/nrwl/nx/blob/master/packages/web/src/generators/application/application.spec.ts#L482

Reply:
I hadn't seen the already existing default value but now I guess my question is:

Should we replace
```
 if (options.bundler === 'vite' && !options.unitTestRunner) {
    options.unitTestRunner = 'vitest';
  }
 // ...
  options.unitTestRunner = options.unitTestRunner || 'jest';
```

with
```
if (!options.unitTestRunner) {
  options.unitTestRunner = options.bundler === 'vite' ? 'vitest' : 'jest';
}
```

Reply:
no because it can still be 'none' in which case it should not change to 'jest'

Reply:
```js
if (!options.unitTestRunner) {
  options.unitTestRunner = options.bundler === 'vite' ? 'vitest' : 'jest';
}
```

has the same behavior as the current code. The only difference is that is "group" the normalization instead of having it handled at 2 different places.

> no because it can still be 'none' in which case it should not change to 'jest'

when it's 'none' the if clause would not execute.